index.js

```diff
@@ -204,6 +214,134 @@ function decorateCollection(collection) {
       }
     }
   };
+
+  // ensureIndex is deprecated but createIndex has exactly the
+  // same behavior
+
+  newCollection.ensureIndex = function(fieldOrSpec, options, callback) {
+    return newCollection.createIndex.apply(newCollection, Array.prototype.slice.call(arguments));
+  };
+
+  newCollection.insert = function(docs, options, callback) {
+    // Use undeprecated equivalents for the two use cases
+    if (Array.isArray(docs)) {
+      return newCollection.insertMany.apply(newCollection, Array.prototype.slice.call(arguments));
+    } else {
+      return newCollection.insertOne.apply(newCollection, Array.prototype.slice.call(arguments));
+    }
+  };
+
+  newCollection.remove = function(selector, options, callback) {
+    // Undeprecated equivalents
+    if (options && ((typeof options) === 'object') && options.single) {
+      arguments[1] = omit(arguments[1], [ 'single' ]);
+      return newCollection.deleteOne.apply(newCollection, Array.prototype.slice.call(arguments));
+    } else {
+      return newCollection.deleteMany.apply(newCollection, Array.prototype.slice.call(arguments));
+    }
+  };
+
+  newCollection.update = function(selector, doc, _options, callback) {
+    var takesCallback = (typeof arguments[arguments.length - 1]) === 'function';
+    var options = (_options && ((typeof _options) === 'object') && _options) || {};
```

Comment:
The last `&& _options` is unnecessary.

Reply:
> That last && _options is there to actually yield it as the value (&&
returns the value of the last thing).

Ah, true. That makes sense, but I think it sacrifices some clarity for negligible terseness. 
`_options && ((typeof _options) === 'object') ? _options : {};` being a touch shorter, even.